runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -284,7 +290,9 @@ protected boolean doesEntryUpdateContext(final ILogData data) {
      */
     private void updatePointer(final ILogData data) {
         // Update the global pointer, if it is non-checkpoint data.
-        if (data.getType() == DataType.DATA && !data.hasCheckpointMetadata()) {
+        if ((data.isData() && !data.hasCheckpointMetadata()) ||
+                (data.isHole() && (data.getBackpointerMap().isEmpty() || data.containsStream(getCurrentContext().id)))
+        ) {
```

Comment:
I think we probably don't need to check containsStream here because when we call this method, logdata that does not belong to this stream should already be filtered out, could you double check?

Reply:
yes, removed.